homeassistant/components/deconz/climate.py

```diff
@@ -68,8 +72,10 @@ def hvac_mode(self):
 
         Need to be one of HVAC_MODE_*.
         """
-        if self._device.on:
-            return HVAC_MODE_HEAT
+        if self._device.mode in SUPPORT_HVAC:
+            return self._device.mode
+        # If unsupported HVAC_MODE then return HVAC_MODE_OFF as default
+        _LOGGER.error("Got an unsupported HVAC_MODE: '%s' mode is %s", self._device.name, self._device.mode)
```

Comment:
We should use;
```
if self._device.on:
             return HVAC_MODE_HEAT
```
As a second control rather than having an error

So; check mode or check on or False

Reply:
From what I learned from my testings so far, for the Eurotronic Spirit, self._device.on is always true. The relevant property to look at is self._device.state_on (I don't know if this is the same for other Thermostats devices). But this property is True when the thermostat is in heat and auto mode, so we can't distinguish them. I can add a test on this property, but I wonder then if it is better to return HVAC_MODE_HEAT or HVAC_MODE_AUTO by default?

I have absolutely no idea if those 3 modes (heat, auto and off) will always be supported. Anyway, I don't know how to get supported features from deConz (I haven't seen an API call to get this information), and so far Eurotronic seems to be the only supported thermostat (I thought I saw a page on the web mentioning another one, but I can't find it again).

And thank you very much indeed for reviewing my PR (this is my first one on GitHub, so I am discovering the process, my apology if I am not doing everything in the state of the art).